### PR TITLE
refactor: remove DB generic

### PR DESCRIPTION
**Pull request by bsbds**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

  The DB generic is redundant as no other type will ever implement `StorageApi` in Xline. This change aims to enhance readability of the codebase.

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Comment by bsbds** [comment]
> Though xline currently has only one StorageApi impl，the generic is make sense for the community if someone want to use another storage engine.

The `engine` crate already provides a level of abstraction, I think introducing the `StorageApi` would be redundant.

File: crates/xline/src/storage/compact/mod.rs [hunk 1]
```diff
@@ -97,16 +96,15 @@ pub(crate) async fn auto_compactor<C: Compactable>(
 
 /// background compact executor
 #[allow(clippy::arithmetic_side_effects, clippy::ignored_unit_patterns)] // introduced bt tokio::select! macro
-pub(crate) async fn compact_bg_task<DB>(
-    kv_store: Arc<KvStore<DB>>,
+#[allow(clippy::arithmetic_side_effects)] // introduced bt tokio::select! macro
```
Comment (Phoenix500526) [inline crates/xline/src/storage/compact/mod.rs:99]: Please remove this duplicated `#[allow(clippy::arithmetic_side_effects)]`

File: crates/xline/src/server/xline_server.rs [hunk 2]
```diff
@@ -434,20 +433,20 @@ impl XlineServer {
     #[allow(clippy::type_complexity, clippy::too_many_lines)] // it is easy to read
     #[allow(clippy::as_conversions)] // cast to dyn
     #[allow(trivial_casts)] // same as above
```
Comment (Phoenix500526) [inline crates/xline/src/server/xline_server.rs:435]: Please merge these three lines into one.

File: crates/xline/src/server/maintenance.rs [hunk 3]
```diff
@@ -32,46 +32,40 @@ const MIN_PAGE_SIZE: u64 = 512;
...
+pub(crate) struct MaintenanceServer {
     /// Kv Storage
-    kv_store: Arc<KvStore<S>>,
+    kv_store: Arc<KvStore>,
     /// Auth Storage
-    auth_store: Arc<AuthStore<S>>,
+    auth_store: Arc<AuthStore>,
     /// persistent storage
-    persistent: Arc<S>, // TODO: `persistent` is not a good name, rename it in a better way
+    persistent: Arc<DB>, // TODO: `persistent` is not a good name, rename it in a better way
```
Comment (Phoenix500526) [inline crates/xline/src/server/maintenance.rs:41]: Please rename this field.

**Review by rogercloud — COMMENTED** [review]
If we remove the DB generic here, is it necessary to keep `StorageApi` trait?

File: crates/xline/src/storage/compact/mod.rs [hunk 4]
```diff
@@ -97,16 +96,15 @@ pub(crate) async fn auto_compactor<C: Compactable>(
 
 /// background compact executor
 #[allow(clippy::arithmetic_side_effects, clippy::ignored_unit_patterns)] // introduced bt tokio::select! macro
```
[the remark this answers is not in the document]
Comment (bsbds) [reply crates/xline/src/storage/compact/mod.rs]: fixed

**Comment by bsbds** [comment]
> If we remove the DB generic here, is it necessary to keep `StorageApi` trait?

removed

File: crates/xline/src/server/maintenance.rs (hunk 3 above)
Comment (bsbds) [reply crates/xline/src/server/maintenance.rs]: done

**Review by Phoenix500526 — APPROVED** [review]
LGTM. Please squash these commits into one commit

**Review by CrystalAnalyst — APPROVED** [review]
LGTM.

**Pull request merged**